From 8079fa6bd4754479a906ffd5ec6bc7aa43b30656 Mon Sep 17 00:00:00 2001
From: Gaurav Vohra <gauravx.vohra@intel.com>
Date: Mon, 15 Apr 2019 17:21:18 +0800
Subject: [PATCH] ASoC: TDF8532: Fix for double memory free in case of error
 condition of tdf8523_single_read

tdf8523_single_read is defind to return uint8_t while the return statement,
returns a int type, so in case of error checks are not followed (E.g -121 is returned as 135)
Which lead to double free of memory as its already freed in tdf8532_single_read in case of
failure.

Signed-off-by: Gaurav Vohra <gauravx.vohra@intel.com>
---
 sound/soc/codecs/tdf8532.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/codecs/tdf8532.c b/sound/soc/codecs/tdf8532.c
index 65d9b1c..b2aea89 100644
--- a/sound/soc/codecs/tdf8532.c
+++ b/sound/soc/codecs/tdf8532.c
@@ -107,11 +107,11 @@ static uint8_t tdf8532_read_wait_ack(struct tdf8532_priv *dev_data,
 	return ret;
 }
 
-static uint8_t tdf8532_single_read(struct tdf8532_priv *dev_data,
+static int tdf8532_single_read(struct tdf8532_priv *dev_data,
 					char **repl_buff)
 {
 	struct device *dev = &(dev_data->i2c->dev);
-	uint8_t recv_len;
+	int recv_len;
 	int ret;
 
 	ret = tdf8532_read_wait_ack(dev_data, msecs_to_jiffies(ACK_TIMEOUT));
-- 
1.9.1

